torcx/what-is-torcx.md

```diff
@@ -0,0 +1,31 @@
+# What is Torcx?
+
+[Torcx][gh-torcx] is a new boot-time addon manager designed specifically for Container Linux. At the most basic level, it is a tool for applying ephemeral changes to an immutable system during early boot. This includes providing third-party binary addons and installing systemd units, which can vary across environments and boots. On every boot, Torcx reads its configuration from local disk and propagates specific assets provided by addon packages (which must be available in local stores).
+
+## Torcx overview
+
+Torcx is meant to be a complementary tool to [Ignition][ignition] and [systemd][systemd], customizing a Container Linux system at runtime without having to build a custom OS image. This goal is achieved by following two main principles: customizations are ephemeral and they are applied exactly once per-boot. Torcx also has a very simple design, with the aim of providing a small low-level system utility which can be driven by more advanced and higher-level tools.
+
+### Torcx execution model and systemd generators
+
+Early in the boot process, execution starts in a minimal initramfs environment where systemd, Ignition and other helpers run. Once up, execution continues by pivoting into the real root file-system and by running all [systemd generators][systemd-generator], including the main torcx component called `torcx-generator`. This is run serially before any other services are started; for this reason, it is guaranteed not to race with any other startup logic but at the same time it cannot rely on other non-local resources, including remote file-systems and assets reachable over the network.
+
+### Profiles and addons
+
+Torcx customizations are applied via local addon packages, which are referenced by profiles. Addons are simple tar-gzipped archives containing binary assets and a manifest An user profile (upper profile) can be supplied by the administrator to be merged on top of hard-coded vendor and OEM profiles (lower profiles), and torcx will take care of applying the resulting list of addons on the systems.
+
+### Boot-time customizations
+
+By design Torcx guarantees that customizations are applied at most once per boot, before any other service has been considered for startup. This is done in order to provide a mechanism to customize most aspects of a Container Linux system in a reliable way, and to avoid runtime upgrading/downgrading issues. Changes applied via Torcx are not persisted to disk, in order to last exactly for the lifetime of a single boot of an instance.
+
+Conversely, this means that Torcx should not be seen and abused as a general container runtime or package manager, as additional services provided this way result in an increased memory usage and requires a system reboot for upgrades/downgrades.
```

Comment:
Replace with: "By the same token, this should be read as a warning against abusing Torcx in the role of a general purpose container, service, or package manager. Torcx's boot-transient model consumes memory with each addon, and, worse, would require system reboots for even simple upgrades."